### PR TITLE
[Silabs] Si917: C files converted to C++.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
C files converted to C++:
* SiWx917/sl_wifi_if.c, 
* SiWx917/wfx_rsi_host.c

Compiled in BRD4338A.

### CODE REVIEW

File: examples/platform/silabs/efr32/rs911x/sl_wifi_if.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+../../SiWx917/SiWx917/sl_wifi_if.c
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/sl_wifi_if.cpp:1]: Broken symlink ? if the file was change for .cpp extension

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+{
+    sl_wfx_mac_address_t * mac;
+
+#ifdef SL_WFX_CONFIG_SOFTAP
+    mac = (interface == SL_WFX_SOFTAP_INTERFACE) ? &wfx_rsi.softap_mac : &wfx_rsi.sta_mac;
+#else
+    mac = &wfx_rsi.sta_mac;
+#endif
+    *addr = *mac;
+    SILABS_LOG("%s: %02x:%02x:%02x:%02x:%02x:%02x", __func__, mac->octet[0], mac->octet[1], mac->octet[2], mac->octet[3],
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:117]: This log can be remove. Printing the Mac address every time we call the get_wifi_mac_addr fct seems like redondant. Logs if any should be in the caller function, not in the getter...

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+void wfx_get_wifi_mac_addr(sl_wfx_interface_t interface, sl_wfx_mac_address_t * addr)
+{
+    sl_wfx_mac_address_t * mac;
+
+#ifdef SL_WFX_CONFIG_SOFTAP
+    mac = (interface == SL_WFX_SOFTAP_INTERFACE) ? &wfx_rsi.softap_mac : &wfx_rsi.sta_mac;
+#else
+    mac = &wfx_rsi.sta_mac;
+#endif
+    *addr = *mac;
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:116]: No nullptr verification before dereferencing.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+ *      Driver set the wifi provision
+ * @param[in]  cfg:  wifi configuration
+ * @return
+ *       None
+ ***********************************************************************/
+void wfx_set_wifi_provision(wfx_wifi_provision_t * cfg)
+{
+    SILABS_LOG("%s: SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
+
+    wfx_rsi.sec = *cfg;
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:133]: No nullptr verification, should return immediatly if the struc is a `nullptr`

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+ * @fn  void wfx_set_wifi_provision(wfx_wifi_provision_t *cfg)
+ * @brief
+ *      Driver set the wifi provision
+ * @param[in]  cfg:  wifi configuration
+ * @return
+ *       None
+ ***********************************************************************/
+void wfx_set_wifi_provision(wfx_wifi_provision_t * cfg)
+{
+    SILABS_LOG("%s: SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:131]: Log can be remove

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+sl_status_t wfx_connect_to_ap(void)
+{
+    if (wfx_rsi.dev_state & WFX_RSI_ST_STA_PROVISIONED)
+    {
+        SILABS_LOG("%s: connecting to access point -> SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
+        xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+    }
+    else
+    {
+        SILABS_LOG("%s: error: access point not provisioned", __func__);
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:188]: __func__ can be remove

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+ * Start a JOIN command to the AP - Done by the wfx_rsi task
+ * @param[in]   None
+ * @return  returns SL_STATUS_OK if successful,
+ *         SL_STATUS_INVALID_CONFIGURATION otherwise
+ ****************************************************************************/
+sl_status_t wfx_connect_to_ap(void)
+{
+    if (wfx_rsi.dev_state & WFX_RSI_ST_STA_PROVISIONED)
+    {
+        SILABS_LOG("%s: connecting to access point -> SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:183]: __func__ can be remove

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+ *      called fuction when driver is connected to STA
+ * @param[in]  None
+ * @return  returns ture if successful,
+ *          false otherwise
+ ***********************************************************************/
+bool wfx_is_sta_connected(void)
+{
+    bool status;
+    status = (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED) ? true : false;
+    return status;
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:252]: useless code here cane be optimized 
```suggestion
    return (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED);
```

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+ * @param[in]  which_if:
+ * @return  returns ture if successful,
+ *          false otherwise
+ ***********************************************************************/
+bool wfx_have_ipv6_addr(sl_wfx_interface_t which_if)
+{
+    bool status = false;
+    if (which_if == SL_WFX_STA_INTERFACE)
+    {
+        status = (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED) ? true : false;
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp:322]:
```suggestion
        return (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED);
```

**Review by jepenven-silabs — APPROVED** [review]
Beyond the broken symlink the rest of the comments can be applied in a follow up PR.

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/platform/silabs/efr32/rs911x/sl_wifi_if.cpp (hunk 1 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/sl_wifi_if.cpp]: Symlink fixed.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 2 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Log removed.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 4 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Null pointer checked.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 6 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Function name removed.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 8 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Function name removed.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 9 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Fixed as `status = (wfx_rsi.dev_state & WFX_RSI_ST_STA_DHCP_DONE) > 0;`

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 5 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Removed.

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp (hunk 7 above)
Comment (rcasallas-silabs) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp]: Function name removed.

**Pull request merged**